extensions/httpserver/internal.m

```diff
@@ -51,6 +59,48 @@ - (id)init {
 }
 @end
 
+@implementation HSWebSocket
+
+- (void)didOpen
+{
+    [super didOpen];
+    [LuaSkin logInfo:@"Opened websocket connection"];
+}
+
+- (void)didReceiveMessage:(NSString *)msg
+{
+    __block NSData *response = nil;
+
+    void (^responseCallbackBlock)(void) = ^{
+        LuaSkin *skin = [LuaSkin shared];
+        [skin pushLuaRef:refTable ref:self.callback];
+        lua_pushstring(skin.L, [msg UTF8String]);
+
+        if (![skin protectedCallAndTraceback:1 nresults:1]) {
+            const char *errorMsg = lua_tostring(skin.L, -1);
+            [skin logError:[NSString stringWithFormat:@"hs.httpserver:websocket callback error: %s", errorMsg]];
+        } else {
+            response = [skin toNSObjectAtIndex:-1];
+        }
+    };
```

Comment:
Thanks. I've also got websocket clients in hs.http working, just needs to be cleaned up.